### PR TITLE
Prometheus: Cleanup annotation editor

### PR DESCRIPTION
**Pull request by aocenas**
After: https://github.com/grafana/grafana/pull/48814 there was some angular code left.

Also the switch to builder did not work because of how state is handled right now. So the switch was removed here which allowed to simplify the annotation editor a bit.

### CODE REVIEW

**Comment by yesoreyeram** [comment]
@aocenas - Was testing the annotations but don't see any annotation details. What am I doing wrong here?

* created an annotation with `ALERTS` as query
image
* Dashboard shows the annotation but no details. Field mapping have issues?
image

**Comment by yesoreyeram** [comment]
nvm. As discussed in the chat it require field mappings. ( I thought there could be some defaults )

File: public/app/plugins/datasource/prometheus/components/AnnotationQueryEditor.tsx [hunk 1]
```diff
@@ -19,35 +19,47 @@ export function AnnotationQueryEditor(props: Props) {
...
+          >
+            <AutoSizeInput
+              type="text"
+              aria-label="Set lower limit for the step parameter"
+              placeholder={'auto'}
+              minWidth={10}
+              onCommitChange={(ev) => {
+                onAnnotationChange({
+                  ...annotation,
+                  interval: ev.currentTarget.value,
```
Comment (yesoreyeram) [inline public/app/plugins/datasource/prometheus/components/AnnotationQueryEditor.tsx:55]: Can you confirm `step` and `interval` leads to same context here? Previous code, step property was changed and here `interval ` is being changed. ( aria label reads as step parameter )

**Review by yesoreyeram — APPROVED** [review]
LGTM

File: public/app/plugins/datasource/prometheus/components/AnnotationQueryEditor.tsx (hunk 1 above)
Comment (aocenas) [reply public/app/plugins/datasource/prometheus/components/AnnotationQueryEditor.tsx:55]: ah good catch yeah this should be mapped to `step` in the annotation

**Comment by yesoreyeram** [comment]
@aocenas - Do we need to change the placeholders/labels/title? When I use this first time, I didn't realise this require the field mapping

**Comment by aocenas** [comment]
@yesoreyeram 
> Do we need to change the placeholders/labels/title? When I use this first time, I didn't realise this require the field mapping. Or at least we should handle the default scenario i.e, when there is no field mapping provided.

Changed. Not sure what would be good defaults, it depends on what your query is and what labels and even so you choose labels + text that makes sense to you.

**Review by yesoreyeram — APPROVED** [review]

**Pull request merged**